packages/typescript-estree/src/create-program/createSourceFile.ts

```diff
@@ -25,4 +26,11 @@ function createSourceFile(parseSettings: ParseSettings): ts.SourceFile {
       );
 }
 
-export { createSourceFile };
+function createNoProgram(parseSettings: ParseSettings): ASTAndNoProgram {
+  return {
+    ast: createSourceFile(parseSettings),
+    program: null,
+  };
+}
```

Comment:
Nit: I might just skip adding this altogether and have the consumer put `{ ast: createSourceFile(parseSettings) }`. That's a stylistic concern though 🤷 no hard preference.

Reply:
I did it just because it creates a nice symmetry/consistency in our `getProgramAndAST` function.

https://github.com/typescript-eslint/typescript-eslint/pull/6066/files#diff-a8d8fbbee764653ef3a8c2803fff21f3f825a8ad18e8a62cddbe6729555772caR79

```ts
function getProgramAndAST(
  parseSettings: ParseSettings,
  shouldProvideParserServices: boolean,
): ASTAndProgram {
  if (parseSettings.programs) {
    const fromProvidedPrograms = useProvidedPrograms(
      parseSettings.programs,
      parseSettings,
    );
    if (fromProvidedPrograms) {
      return fromProvidedPrograms;
    }
  }

  if (shouldProvideParserServices) {
    const fromProjectProgram = createProjectProgram(parseSettings);
    if (fromProjectProgram) {
      return fromProjectProgram;
    }

    // eslint-disable-next-line deprecation/deprecation -- will be cleaned up with the next major
    if (parseSettings.DEPRECATED__createDefaultProgram) {
      // eslint-disable-next-line deprecation/deprecation -- will be cleaned up with the next major
      const fromDefaultProgram = createDefaultProgram(parseSettings);
      if (fromDefaultProgram) {
        return fromDefaultProgram;
      }
    }

    return createIsolatedProgram(parseSettings);
  }

  // no need to waste time creating a program as the caller didn't want parser services
  // so we can save time and just create a lonesome source file
  return createNoProgram(parseSettings);
}
```

It's just a small style thing rather than the last line being `return { ast: createSourceFile(), program: null }`